projects/packages/jitm/src/class-jitm.php

```diff
@@ -54,12 +47,18 @@ public static function get_instance() {
 	 * Sets up JITM action callbacks if needed.
 	 */
 	public function register() {
-		if ( self::$jitms_registered ) {
+		if ( did_action( 'jetpack_registered_jitms' ) ) {
 			// JITMs have already been registered.
 			return;
 		}
 
-		self::$jitms_registered = true;
+		/**
+		 * Fires when the JITMs are registered. This action is used to ensure that
+		 * JITMs are registered only once.
+		 *
+		 * @since 9.8
+		 */
+		do_action( 'jetpack_registered_jitms' );
```

Comment:
That looks super cool :) Do you think we should do it after adding actions though? Like, maybe other plugins/packages would want to hook on that action and change/remove the actions that JITMs have set?

Reply:
Good point! In commit 32c5ec9, I moved the `do_action` call to the end of the `JITM::register` method.